training/src/main/scala/com/airbnb/aerosolve/training/Evaluation.scala

```diff
@@ -117,12 +117,19 @@ object Evaluation {
         }
         metrics.append((prefix + "ALL_PAIRS_HINGE_LOSS", (hingeLoss, 1.0)))
         var inTopK = false
+        var nAccurateLabel = 0.0
+        var sumPrecision = 0.0
         for (i <- 0 until sorted.size) {
```

Comment:
will need i to do sorted(i)._1 , I think even change it to sorted(i-1)._1 will still need to change the for loop to (i <- 1 until sorted.size + 1), so maybe just keep the a little verbose way